storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -3815,6 +3828,7 @@ public void setBlobMeta(String key, SettableBlobMeta meta)
         throws AuthorizationException, KeyNotFoundException, TException {
         try {
             blobStore.setBlobMeta(key, meta, getSubject());
+            blobStore.updateLastModTime();
```

Comment:
What will happen if `blobStore.setBlobMeta` succeeds then `blobStore.updateLastModTime()` fails? Or nimbus crashed before `blobStore.updateLastModTime()` can run?
 
Looks like supervisors will not be notified for this blob update. We should update the timestamp before doing the actual operation.

Reply:
If the updateLastModTime fails, eventually a Nimbus should get restarted or HDFS fixed, and Nimbus will call blobStore.validateModTime().  This should fix the issue.  During the period in-between, supervisors will remain on an older version of the blob.

Reply:
Thanks. My biggest question is: will the bucketDir's modTime update when the content of a file under the `bucketDir` updates? 
I think your answer is yes.

If that's so, I agree that `validateModTime` will serve the purpose.

Reply:
yes, validateModTime() should update the mod time if a file in the bucket dir is updated.  There will also be occasional false positive updates if a blob is deleted in a bucket dir that contains a different blob file.